crates/red_knot_workspace/src/site_packages.rs

```diff
@@ -69,7 +195,35 @@ fn site_packages_dir_from_sys_prefix(
     //
     // [the non-Windows branch]: https://github.com/python/cpython/blob/a8be8fc6c4682089be45a87bd5ee1f686040116c/Lib/site.py#L401-L410
     // [the `sys`-module documentation]: https://docs.python.org/3/library/sys.html#sys.platlibdir
-    for entry_result in system.read_directory(&sys_prefix_path.join("lib"))? {
+
+    // If we were able to figure out from the `pyvenv.cfg` file what Python version this venv is for,
+    // we should be able to avoid iterating through all items in the `lib/` directory:
+    if let Some(minor_version) = venv_minor_version {
+        let expected_path =
+            sys_prefix_path.join(format!("lib/python3.{minor_version}/site-packages"));
+        if system.is_directory(&expected_path) {
+            return Ok(expected_path);
+        }
+        if minor_version.free_threaded_build_available() {
+            let alternative_path =
+                sys_prefix_path.join(format!("lib/python3.{minor_version}t/site-packages"));
```

Comment:
Uh, the `t` is subtle! I had to compare the strings in my text editor to spot the difference

Reply:
Yes, this is quite annoying... Is there a way of making this more obvious, do you think?

Reply:
I added a comment